addons/l10n_in/models/account_invoice.py

```diff
@@ -73,6 +74,31 @@ def _onchange_name_warning(self):
             }}
         return super()._onchange_name_warning()
 
+    @api.depends('invoice_line_ids.product_id', 'company_id.l10n_in_hsn_code_digit')
+    def _compute_hsn_code_warning(self):
+        for move in self:
+            products = move.invoice_line_ids.filtered(lambda line:
+                            line.product_id.l10n_in_hsn_code and len(line.product_id.l10n_in_hsn_code) < int(move.company_id.l10n_in_hsn_code_digit)
```

Comment:
```suggestion
                            line.l10n_in_hsn_code and len(line.l10n_in_hsn_code) < int(move.company_id.l10n_in_hsn_code_digit)
```

now we have HSN code in lines so check from there.
also message is updated.
```
Ensure that the HSN/SAC Code consists of at least 6 digits in lines 2 and 5.
```

Reply:
Done